lib/compat.php

```diff
@@ -315,6 +315,40 @@ function gutenberg_register_rest_api_post_type_capabilities() {
 }
 add_action( 'rest_api_init', 'gutenberg_register_rest_api_post_type_capabilities' );
 
+/**
+ * Includes the value for the 'previewable' attribute of a post type resource.
+ *
+ * @see https://core.trac.wordpress.org/ticket/43739
+ *
+ * @param object $post_type Post type response object.
+ * @return boolean Whether or not the post type can be previewed.
+ */
+function gutenberg_get_post_type_previewable( $post_type ) {
+	return is_post_type_viewable( $post_type['slug'] );
+}
+
+/**
+ * Adds the 'previewable' attribute to the REST API response of a post type.
+ *
+ * @see https://core.trac.wordpress.org/ticket/43739
+ */
+function gutenberg_register_rest_api_post_type_previewable() {
+	register_rest_field( 'type',
+		'previewable',
```

Comment:
Is there are particular reason for calling it "previewable", rather than "viewable"? I added a similar flag in this PR: https://github.com/WordPress/gutenberg/pull/5756/files#diff-6ff32417da0658502e7caa1a1abbeae6R403

But "previewable" is not quite the flag name I'd be using in that case.

Reply:
No particular reason. Changed it to `viewable`.

Reply:
Noice. I shamelessly stole this and merged it in 4c90150, so it doesn't need to be in this PR anymore.